### PR TITLE
Feature/coverage

### PR DESCRIPTION
**Pull request by idodeclare**
This is just a patch to add test coverage for many of the zeroes in coveralls.

### CODE REVIEW

File: opengrok-web/src/test/java/org/opengrok/web/util/FileUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.Test;
+import java.io.FileNotFoundException;
+import java.util.UUID;
+
+/**
+ * Represents a container for tests of {@link FileUtil}.
+ */
+public class FileUtilTest {
+
+    @Test
```
Comment (vladak) [inline opengrok-web/src/test/java/org/opengrok/web/util/FileUtilTest.java:37]:
```suggestion
    @Test(expected = NoPathParameterException.class)
```

Comment (vladak) [reply opengrok-web/src/test/java/org/opengrok/web/util/FileUtilTest.java:37]: or given that we already use Junit 5 it should be possible to use `assertThrows()`

File: plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    @Before
+    public void setUp() {
+        plugin = new UserWhiteListPlugin();
+    }
+
+    @Test
+    public void shouldThrowOnLoadIfNullArgument() {
+        NullPointerException caughtException = null;
```
Comment (vladak) [inline plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java:86]: ditto - use JUnit mechanisms for asserting the exceptions ?

File: plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    public static void beforeClass() throws Exception {
+        tempWhitelist = File.createTempFile("UserWhiteListPluginTest", "txt");
+        try (BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+                new FileOutputStream(tempWhitelist), StandardCharsets.UTF_8))) {
+            writer.write(OK_USER);
+            // Don't bother with trailing LF.
+        }
+
+        validPluginParameters = new HashMap<>();
+        validPluginParameters.put(UserWhiteListPlugin.FILE_PARAM, tempWhitelist.getPath());
```
Comment (vladak) [inline plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java:68]: would be nice to see what happens if the parameters include path to non-existent file.

**Comment by vladak** [comment]
This looks good, I only had a few nits. Please rebase.

File: plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        plugin.load(validPluginParameters);
+
+        DummyHttpServletRequest req = new DummyHttpServletRequest();
+        req.setAttribute(UserPlugin.REQUEST_ATTR, new User(OK_USER));
+
+        Project randomProject = new Project(RandomString.generateUpper(10));
+        boolean projectAllowed = plugin.isAllowed(req, randomProject);
+        assertTrue("should allow OK_USER for random project 1", projectAllowed);
+
+        randomProject = new Project(RandomString.generateUpper(10));
```
Comment (vladak) [inline plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java:127]: why test 2 projects if they are both random ?

File: opengrok-web/src/test/java/org/opengrok/web/util/FileUtilTest.java (hunk 1 above)
Comment (idodeclare) [reply opengrok-web/src/test/java/org/opengrok/web/util/FileUtilTest.java:37]: Done

File: plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java (hunk 3 above)
Comment (idodeclare) [reply plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java:69]: Done

**Comment by vladak** [comment]
Thanks !

**Pull request merged**